# MOVIES ANALYSIS WITH EDA

## BUSINESS UNDERSTANDING
### What factors should Microsoft consider when starting a Movie Studio?
Microsoft has decided to start a new Movie Studio. In order to do this they need to know the kinds of movies that are doing the best.This will help to know the type of films that microsoft should produce.EDA should be done so as to provide concrete recommendations to Microsoft.

## Data Understanding

The data for this analysis comes from;
* ### im.db
This is a SQLite database that consists of data about movies from IMDB.

* ### bom.movie_gross.csv
This is a csv file that contains data about movies from Box Office

## Importing Libraries

In [1]:
import numpy as np

#for analysis and manipulation of data
import pandas as pd

#for connecting to im.db
import sqlite3 

# for visualization
import matplotlib.pyplot as plt
import seaborn as sns

## Loading Data

In [2]:
# connecting to im.db database
imdb_df = sqlite3.connect("movies-Data/im.db")

In [3]:
#load the bom data
bom_df = pd.read_csv("movies-Data/bom.movie_gross.csv")

## Data Understanding (Box Office)

In [4]:
# preview of first 5 rows
bom_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [5]:
# preview of the last rows
bom_df.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [6]:
# Number of rows and columns
bom_df.shape

(3387, 5)

In [7]:
#Statistical summary of data
bom_df.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [8]:
# information about the columns
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This is what we get from the above information;
* Box Office data has 5 columns namely ```title```, ```studio```, ```domestic_gross```, ```foreign_gross``` and ```year```. 
* ```title```, ```studio``` and ```foreign_gross``` columns are string ojects while ```domestic_gross``` and ```year``` are numerical objects.
* The data has a total of 3387 rows.
* The following columns have null values because they don't have all 3387 rows;
    * studio
    * domestic_gross
    * foreign_gross

## Data Cleaning(Box Office)


In [9]:
# number of rows with null values
bom_df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [10]:
# sample rows where foreign_gross has no missing values
bom_df[bom_df["studio"].notna()].sample(5, random_state = 1)

title  studio  domestic_gross  \
571                                 Poetry    Kino        356000.0   
3343                      The Third Murder      FM         89300.0   
1850                            Camp X-Ray     IFC         13300.0   
3257  Don't Worry He Won't Get Far on Foot  Amazon       1400000.0   
1512                              Non-Stop    Uni.      92200000.0   

     foreign_gross  year  
571        1900000  2011  
3343           NaN  2018  
1850           NaN  2014  
3257       2500000  2018  
1512     130600000  2014

In [11]:
# sample rows where foreign_gross has missing values
bom_df[bom_df["studio"].isna()]

title studio  domestic_gross foreign_gross  year
210   Outside the Law (Hors-la-loi)    NaN         96900.0       3300000  2010
555         Fireflies in the Garden    NaN         70600.0       3300000  2011
933           Keith Lemon: The Film    NaN             NaN       4000000  2012
1862                 Plot for Peace    NaN          7100.0           NaN  2014
2825               Secret Superstar    NaN             NaN     122000000  2017

In [12]:
# drop rows where studio has a null value
bom_df.dropna(subset = ["studio"], inplace = True)

In [13]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   2033 non-null   object 
 4   year            3382 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 158.5+ KB


In [14]:
# percentage of missing values in domestic_gross
bom_df["domestic_gross"].isnull().sum()/len(bom_df)*100

0.768775872264932

This is a very small portion of the dataset therefore we can just drop the rows with null values.

In [15]:
bom_df.dropna(subset = ["domestic_gross"], inplace = True)
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3356 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3356 non-null   object 
 1   studio          3356 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   2007 non-null   object 
 4   year            3356 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 157.3+ KB


In [16]:
# percentage of missing values in foreign gross
bom_df["foreign_gross"].isnull().sum()/len(bom_df)*100

40.19666269368295

This is a significant portion of the dataset therefore we cannot just drop the rows. We can fill the missing values with either the mean or median.

In [17]:
# finding the mean before replacing the null values
bom_df["foreign_gross"].head()

0    652000000
1    691300000
2    664300000
3    535700000
4    513900000
Name: foreign_gross, dtype: object

Looks like the datatype of foreign gross is string. So we need to change the column to a numerical datatype.

In [22]:
#remove whitespaces
bom_df.columns = bom_df.columns.str.strip()

# convert to a numerical datatype
bom_df["foreign_gross"] = bom_df["foreign_gross"].astype(pd.Int32Dtype())

TypeError: object cannot be converted to an IntegerDtype

In [135]:
#median of domestic_gross
bom_df["foreign_gross"].median()

TypeError: 

In [52]:
#checking mean after filling null values with the mean
bom_df1 = bom_df["domestic_gross"].fillna(bom_df["domestic_gross"].mean())
bom_df1.mean().round(2)

28771489.56

In [53]:
#checking mean after filling null values with median
bom_df2 = bom_df["domestic_gross"].fillna(bom_df["domestic_gross"].median())
bom_df2.mean().round(2)

28561064.16

According to the above results, filling null values with the median is not a good option because it reduces the mean. So we will fill the null values with the mean because that  does not alter the mean.

In [54]:
# replacing null values with the mean
bom_df["domestic_gross"] = bom_df["domestic_gross"].fillna(bom_df["domestic_gross"].mean())
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3382 non-null   float64
 3   foreign_gross   2033 non-null   object 
 4   year            3382 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 158.5+ KB


In [60]:
# percentage of missing values in foreign_gross
(bom_df["foreign_gross"].isna().sum()/(len(bom_df))*100)

39.8876404494382